nexus/src/app/mod.rs

```diff
@@ -58,6 +61,27 @@ pub(crate) const MAX_NICS_PER_INSTANCE: usize = 8;
 // TODO-completness: Support multiple external IPs
 pub(crate) const MAX_EXTERNAL_IPS_PER_INSTANCE: usize = 1;
 
+pub(crate) struct ExternalServers {
+    config: dropshot::ConfigDropshot,
+    https_port: u16,
+    http: Option<DropshotServer>,
```

Comment:
Is the `http` server actually optional (given `refresh_tls_config` can fail if it's `None`)?

Reply:
`refresh_tls_config` really just wants to get the `ServerContext` from *somewhere* - I refactored this to abff7032de80f20fb2f4020ed917587d895f74d4.

For now: Yes, the `http` server is optional, but that's kinda because of the weird relationship between `ServerContext` and `Nexus`. Namely: `Nexus` is created first (without servers - this is when it truly must be optional), then `ServerContext` is created (referencing `Nexus`), then `Nexus`' servers can be initialized (referencing `ServerContext`, making it non-optional).

I could make it `OnceCell` or something, but I actually do want to eventually allow the http server to be optional, so we can roll with HTTPS only when we have an automated certificate refresh mechanism that's well-tested.

Reply:
Makes sense. I was going to ask if `ExternalServers` (or maybe `Nexus`?) could hold a non-optional `ServerContext`, since we always know we have a single one of those regardless of whether we have an http server / https server / both / neither, but I think that has the same chicken-and-egg problem of creation order and would create an `Arc` cycle (`Nexus` -> `ExternalServers` -> `ServerContext` -> `Nexus`) that would cause all the types involved to never be dropped. The current code has the same cycle, but it's explicitly broken in `close_servers` when we `take()` the servers out of `http`/`https`.

Reply:
To aid my understanding: can both `http` and `https` be `Some` at any given time?

Reply:
Yes, it's possible to have both HTTP and HTTPS servers exposed (on different ports).